### PR TITLE
Check nullability of fruit and vegetables

### PR DESCRIPTION
**Pull request by msalarab**
fix fake data, fix and improve excute function

### CODE REVIEW

File: src/main/kotlin/interactor/GetAverageFruitAndVegetablesInteractor.kt [hunk 1]
```diff
@@ -1,37 +1,28 @@
...
-    fun execute(): List<String> {
+
+    fun execute(limit: Int): List<String> {
         return dataSource
             .getAllCitiesData()
-            .filter(::excludeNullAndNonNumberSalaries)
-            .sortedBy(::lowestCostComparingToTheSalaries)
-            .take(10)
-            .map { it.cityName }
+            .filter(::excludeNullFruitAndVegetablePriceAndNullSalaries)
```
Comment (TarekIdrees) [inline src/main/kotlin/interactor/GetAverageFruitAndVegetablesInteractor.kt]: check out the "And" word may lead to missing of the meaning

File: src/main/kotlin/interactor/GetAverageFruitAndVegetablesInteractor.kt [hunk 2]
```diff
@@ -1,37 +1,28 @@
...
+            apples1kg != null && banana1kg != null && oranges1kg != null && tomato1kg != null
+                    && potato1kg != null && onion1kg != null && lettuceOneHead != null
+        }
 
-    private fun lowestCostComparingToTheSalaries(city: CityEntity): Double {
-        return city.fruitAndVegetablesPrices.getAverageFruitVegetablesPrice()
-            .div(city.averageMonthlyNetSalaryAfterTax!!)
-    }
+    private fun calculateFruitVeggiePriceToSalaryRatio(city: CityEntity) =
+        city.fruitAndVegetablesPrices.getAverageFruitsAndVegetablesPrice() / city.averageMonthlyNetSalaryAfterTax!!
```
Comment (TarekIdrees) [inline src/main/kotlin/interactor/GetAverageFruitAndVegetablesInteractor.kt:26]: I think the function name could be "calculate the ratio between FruitVeg price and salary"

**Review by Ali873-debug — DISMISSED** [review]

**Review by TarekIdrees — DISMISSED** [review]
GJ <3

**Review by mosayed01 — DISMISSED** [review]
💯

**Review by AmeerAmjed — DISMISSED** [review]

**Review by Sajadio — DISMISSED** [review]
good work

**Review by AmeerAmjed — APPROVED** [review]

**Pull request merged**